packages/medusa/src/strategies/price-selection.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  IPriceSelectionStrategy,
+  PriceSelectionContext,
+  PriceSelectionResult,
+  PriceType,
+} from "../interfaces/price-selection-strategy"
+import { MedusaError } from "medusa-core-utils"
+import { MoneyAmountRepository } from "../repositories/money-amount"
+import { EntityManager } from "typeorm"
+
+class PriceSelectionStrategy implements IPriceSelectionStrategy {
```

Comment:
nit: `extends AbstractPriceSelectionStrategy` instead.

Technically it is not necessary when registering the Strategy in the core, however, rather than using the `isPriceSelectionStrategy` type-guard in the plugins loader I actually think it will be better to use something like :
```
if (loaded instanceof AbstractPriceSelectionStrategy)
```

My reasoning for having this updated is that if someone is trying to override the price selection strat it will be nice for them to be able to use this file as a reference for their own implementation and in that way using `extends Abstract...` will reduce some friction.